packages/api/src/major/major-collator.ts

```diff
@@ -0,0 +1,118 @@
+import { Major2 } from "@graduate/common";
+
+const MAJORS: Record<string, Record<string, Major2>> = {};
+const MAJOR_YEARS = new Set<string>();
+
+const rootDir = "./src/major/majors";
+
+interface YearData {
+  year: string;
+}
+
+interface YearCollegeData {
+  year: string;
+  college: string;
+}
+
+interface YearCollegeMajorData {
+  year: string;
+  college: string;
+  major: string;
+}
+
+async function fileExists(
+  fs: typeof import("fs/promises"),
+  path: string
+): Promise<boolean> {
+  return await fs.access(path, fs.constants.F_OK).then(
+    () => true,
+    () => false
+  );
+}
+
+// TODO: this code is quick and dirty but works. this should be replaced with some dry-er code later.
+async function collateMajors() {
```

Comment:
My brain is slow but from what it looks like, we run through each year, each college and add all the majors. initial.json is our initial scrape and commit.json is the hand validated stuff?

If this is the case I can update this with comments/put it in our notion somewhere for knowledge transfer.

Reply:
Good call! Added some comments to the function.